### PR TITLE
Secret resource

### PR DESCRIPTION
**Pull request by tomahawk28**
Mostly follow https://github.com/giantswarm/operatorkit/pull/288

- To support generic `secret` resource operator kit.

### CODE REVIEW

**Comment by tomahawk28** [comment]
After updating the vendor, I failed in an integration test. Looks like we have some issues with delay.
```
# command-line-arguments
controller/integration/test/statusupdate/resource.go:33:32: undefined: invalidConfigError
controller/integration/test/statusupdate/resource.go:79:19: undefined: "github.com/giantswarm/operatorkit/vendor/github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1".NodeConfigStatusCondition
```

**Comment by tomahawk28** [comment]
@xh3b4sd `NodeConfigStatusCondition` was gone by this, https://github.com/giantswarm/apiextensions/pull/194 maybe you know better how to cope with this?

**Comment by xh3b4sd** [comment]
I will have a look and provide a fix.

File: resource/secret/create.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package secret
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	v1 "k8s.io/api/core/v1"
```
Comment (rossf7) [inline resource/secret/create.go]:
```suggestion
	corev1 "k8s.io/api/core/v1"
```

This needs updating in multiple places.

**Comment by xh3b4sd** [comment]
See https://github.com/giantswarm/operatorkit/pull/293 for the fix.

**Comment by xh3b4sd** [comment]
Merged. CI should go green now.

**Pull request merged**